### DISCUSSION TITLE
Import local modules from a webpack 5 bundled library

### DISCUSSION THREAD

**Discussion by doberkofler** [webpack/webpack#14826] [Q&A]
My application uses webpack v5 to bundle all resources. I’m importing from a 3rd party library that itself is also bundled bundled using webpack v5. To customise the 3rd party library, I would now need to access some of the (not exported) local modules within the bundle of the 3rd party library. I fully understand, that this contradicts the intention of keeping modules private, but is there some hack to access the local symbols of a 3rd partly library bundled using webpack?

**Comment by alexander-akait** [comment]
It can be unsafe, do you need access on bundle level or on code level?

**Reply by doberkofler** [reply]
I need to access modules in the already bundled library.
What safety issues do you see, if possible?

**Reply by alexander-akait** [reply]
For example - tree-shaking, author can change your code and some modules will be disappeared, can you provide more information, maybe be small example usage?

**Reply by doberkofler** [reply]
I have several very specific use cases.

Let's assume:

I use a library named `main_lib` that that contains a module `main_mod`.
The module `main_mod` imports the function 'sub_func' from another library `sub_lib`.
The library `main_lib` exports the function `main_func` from the module `main_mod`.

My code imports the function `sub_func` from the library `main_lib`.
My code also needs the function 'sub_func' from the same library `sub_lib` as the library `main_lib`.

I clearly can import from `sub_lib` again but this would duplicate the code and make it difficult to sync the versions.

**Reply by alexander-akait** [reply]
Maybe better to ask developer publish not bundled version

**Reply by doberkofler** [reply]
Already done, but not possible.
My questions is:  is it possible to use the local modules in a bundled webpack 5 library?

**Reply by alexander-akait** [reply]
webpack rewrite own bundled code so you can try https://webpack.js.org/api/module-variables/#__webpack_modules__-webpack-specific

**Reply by doberkofler** [reply]
Where can I find those module variables?

**Reply by alexander-akait** [reply]
It is global variable, put it in your code

**Reply by doberkofler** [reply]
Sorry, I still don't understand. What should I exactly do with `__webpack_modules__`?

**Reply by alexander-akait** [reply]
It is global object which contains all modules, just check it

**Reply by doberkofler** [reply]
I see. webpack injects this object when building. This works now and I actually see each imported module including the library. Unfortunately I do not see the the modules within the external library or do I miss something?

**Reply by alexander-akait** [reply]
So webpack doesn't handle it, it means it is impossible, sorry
